fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4

```diff
@@ -178,6 +178,7 @@ booleanExpression
 
 predicate
     : NOT? kind=BETWEEN lower=valueExpression AND upper=valueExpression
+    | NOT? kind=(LIKE | REGEXP) pattern=valueExpression
```

Comment:
we have RLIKE in DorisLexer.g4 represent both 'RLIKE' and 'REGEXP'

Reply:
DORIS original optimizer does not support RLIKE, the new optimizer should be consistent with it